internal/otelcollector/config/metric/gateway/config.go

```diff
@@ -39,6 +46,37 @@ type KymaStatsReceiver struct {
 	Modules            []ModuleGVR `yaml:"modules"`
 }
 
+type SingletonK8sClusterReceiver struct {
+	K8sClusterReceiver K8sClusterReceiver `yaml:"k8s_cluster"`
+}
+
+type K8sClusterReceiver struct {
+	AuthType                 string                  `yaml:"auth_type"`
+	CollectionInterval       string                  `yaml:"collection_interval"`
+	NodeConditionsToReport   []string                `yaml:"node_conditions_to_report"`
+	AllocatableTypesToReport []string                `yaml:"allocatable_types_to_report"`
+	Metrics                  K8sClusterMetricsConfig `yaml:"metrics"`
+}
+
+type MetricConfig struct {
+	Enabled bool `yaml:"enabled"`
+}
+
+type K8sClusterMetricsConfig struct {
+	// metrics allows enabling/disabling scraped metric.
+	K8sContainerStorageRequest          MetricConfig `yaml:"k8s.container.storage_request"`
+	K8sContainerStorageLimit            MetricConfig `yaml:"k8s.container.storage_limit"`
+	K8sContainerEphemeralStorageRequest MetricConfig `yaml:"k8s.container.ephemeralstorage_request"`
+	K8sContainerEphemeralStorageLimit   MetricConfig `yaml:"k8s.container.ephemeralstorage_limit"`
+	K8sContainerRestarts                MetricConfig `yaml:"k8s.container.restarts"`
+	K8sContainerReady                   MetricConfig `yaml:"k8s.container.ready"`
+	K8sNamespacePhase                   MetricConfig `yaml:"k8s.namespace.phase"`
+	OpenShiftAppliedClusterQuotaLimit   MetricConfig `yaml:"openshift.appliedclusterquota.limit"`
```

Comment:
Do we need to define the openshift metrics?
They will only be emitted if the `distribution` is set as `openshift`